Assets/HotmaxEditorVR/universalTransform.cs

```diff
@@ -0,0 +1,225 @@
+    }
+
+    void updateRotationModeEvent(stateManager.rotationModes value)
+    {
+        _rotationMode = value;
+        if (_rotationGizmoIsSelected)
+        {
+            if (_rotationMode == stateManager.rotationModes.xRotationMode)
+            {
+                setOneRotationGizmoVisible(0);
+            }
+            if (_rotationMode == stateManager.rotationModes.yRotationMode)
+            {
+                setOneRotationGizmoVisible(1);
+            }
+            if (_rotationMode == stateManager.rotationModes.zRotationMode)
+            {
+                setOneRotationGizmoVisible(2);
+            }
+        }
+    }
+
+    void updateEditorMode(stateManager.editorModes value)
+    {
+        _editorMode = value;
+    }
+
+    void setOneRotationGizmoVisible(int gizmo)
+    {
+        Transform xGizmo = init.rotationGizmos.transform.Find("xRotationGizmo");
+        Transform yGizmo = init.rotationGizmos.transform.Find("yRotationGizmo");
+        Transform zGizmo = init.rotationGizmos.transform.Find("zRotationGizmo");
+
+        transforms.Add(xGizmo);
+        transforms.Add(yGizmo);
+        transforms.Add(zGizmo);
+
+        for (int i = 0; i < transforms.Count; i++)
+        {
+            transforms[i].gameObject.SetActive(false);
+        }
+        transforms[gizmo].gameObject.SetActive(true);
+    }
+
+    void setAllRotationGizmosVisible(bool setVisible)
+    {
+        Transform xGizmo = init.rotationGizmos.transform.Find("xRotationGizmo");
+        Transform yGizmo = init.rotationGizmos.transform.Find("yRotationGizmo");
+        Transform zGizmo = init.rotationGizmos.transform.Find("zRotationGizmo");
+
+        transforms.Add(xGizmo);
+        transforms.Add(yGizmo);
+        transforms.Add(zGizmo);
+
+        for (int i = 0; i < transforms.Count; i++)
+        {
+            transforms[i].gameObject.SetActive(setVisible);
+        }
+    }
+
+    void Update()
+    {
+        // -- Keep class on, but check Store to turn on functionality
+        // -- else block turns off gizmo
+        if (_editorMode == stateManager.editorModes.universalTransformMode)
+        {
+            //Telekinesis Mode 
+            if (_selectedObjectIsActive)
+            {
+                init.rotationGizmos.SetActive(false);
+
+                currentPadYPos = inputManager.device2.GetAxis().y;
+
+                Vector3 offset = inputManager.hand2.transform.forward * (distToController + scrollDistance);
+                transform.position = inputManager.hand2.transform.position + offset;
+
+                if (currentPadYPos > _initialPadYPosition)
+                {
+                    scrollDistance += scrollSpeed;
+                    _initialPadYPosition = currentPadYPos;
+                }
+                if (currentPadYPos < _initialPadYPosition)
+                {
+                    scrollDistance -= scrollSpeed;
+                    _initialPadYPosition = currentPadYPos;
+                }
+            }
+            else
```

Comment:
move ELSE block after both IF blocks

Reply:
I will wrap these in their own functions so the if else stuff is separate and more readable.